internal/provider/cqlauth/cql_auth_data_source.go

```diff
@@ -59,10 +66,11 @@ func DataSourceCQLAuth() *schema.Resource {
 
 func dataSourceCQLAuthRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var (
-		c          = meta.(*scylla.Client)
-		clusterID  = int64(d.Get("cluster_id").(int))
-		dcID, dcOK = d.GetOk("datacenter_id")
-		dns        = d.Get("dns").(bool)
+		c         = meta.(*scylla.Client)
+		clusterID = int64(d.Get("cluster_id").(int))
+		dcName    = d.Get("datacenter").(string)
+		dcID      = d.Get("datacenter_id").(int)
```

Comment:
What would happen if there was (or would be) a datacenter with ID == 0? I guess that `d.GetOk` was a good idea here. 

Reply:
For validating the input I would use the (schema.Schema).ValidateFunc:

https://github.com/hashicorp/terraform-plugin-sdk/blob/be3b2d013efe72796bed19b25955bea10d3f8351/helper/schema/schema.go#L349-L342

So not only we can early catch if dcID == 0 but also dcID < 0.

For now all those checks are done server-side.

Reply:
But it would be best to apply such validation throughout the whole codebase, in a separate PR.